### ISSUE TITLE
gatsby-source-mongodb not sanitizing dbName correctly

### ISSUE THREAD

**Issue by raulrpearson** [gatsbyjs/gatsby#11277]
### Description

I tried `gatsby-source-mongodb` with a database name with 2 dashes (something like `one-two-three`) and it seems the name is not properly sanitized. I get the following error when trying to build:

```
Error: Names must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but "mongodbOnetwo-ThreeDocumentsConnectionSortByFieldsEnum" does not.
```

I think the culprit is a missing `g` in the regular expression of the [`sanitizeName`](https://github.com/gatsbyjs/gatsby/blob/cab8d7c53481963ac099d1be427a5600f0d342b4/packages/gatsby-source-mongodb/src/gatsby-node.js#L135) function in that plug in:

```
function sanitizeName(s) {
  return s.replace(/[^_a-zA-Z0-9]/, ``).replace(/\b\w/g, l => l.toUpperCase())
}
```

The first replace stops after the first dash. `s.replace(/[^_a-zA-Z0-9]/, ``)` should be `s.replace(/[^_a-zA-Z0-9]/g, ``)`.

### Steps to reproduce

Change the name of the database in [`using-mongodb`](https://github.com/gatsbyjs/gatsby/blob/cab8d7c53481963ac099d1be427a5600f0d342b4/examples/using-mongodb/gatsby-config.js#L9) and run `gatsby develop`.

### Expected result

The database name `one-two-three` should be sanitized to `Onetwothree` (assuming that was the intention and not OneTwoThree) and the build shouldn't fail.

### Actual result

```
Error: Names must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but "mongodbOnetwo-ThreeDocumentsConnectionSortByFieldsEnum" does not.
```

### Environment

```
  System:
    OS: macOS 10.14.2
    CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
    Shell: 3.0.0 - /usr/local/bin/fish
  Binaries:
    Node: 11.7.0 - /usr/local/bin/node
    Yarn: 1.13.0 - /usr/local/bin/yarn
    npm: 6.6.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  Browsers:
    Chrome: 71.0.3578.98
    Firefox: 64.0.2
    Safari: 12.0.2
  npmPackages:
    gatsby: ^2.0.0 => 2.0.97
    gatsby-plugin-react-helmet: ^3.0.0 => 3.0.5
    gatsby-source-mongodb: ^2.0.5 => 2.0.10
    gatsby-transformer-remark: ^2.1.1 => 2.2.0
  npmGlobalPackages:
    gatsby-cli: 2.4.8
```

**Comment by sidharthachatterjee** [comment]
Thank you for opening this and for debugging this @raulrpearson 

Seems like an easy fix! Would you be interested in opening a PR? We'd love to accept one 🙂

**Comment by harshil1712** [comment]
Yeah I would like to take this one

**Comment by sidharthachatterjee** [comment]
Awesome, thanks @harshil1712

**Comment by raulrpearson** [comment]
> Would you be interested in opening a PR? We'd love to accept one

Thanks, @sidharthachatterjee. Would have loved to submit a PR, but it seems you have to move quickly here 😂 . Maybe next time.

**Comment by raulrpearson** [comment]
Not stale. Will be solved when #11294 is merged.
